rewardbench/models/slicpairpm.py

```diff
@@ -0,0 +1,208 @@
+from dataclasses import dataclass, field
+from typing import Optional
+import torch
+from datasets import load_dataset
+from tqdm import tqdm
+from transformers import AutoTokenizer, HfArgumentParser, pipeline, AutoModelForCausalLM
+import numpy as np
+import pandas as pd
+from typing import List, Optional, Tuple, Union
+
+
+class SlicPairPMPipeline:
+
+    def __init__(self, model_path):
```

Comment:
Also, this needs to be modified to match the loading in the scripts. 
See https://github.com/allenai/reward-bench/blob/a7cf68b5923d42cc16f3a1fa2d9b2b863d140866/scripts/run_rm.py#L173

Mostly need to take in the args, and if they are not all used that's also fine.

Reply:
Modified accordingly. But since it needs an additional tokenizer to prepare the pair (x, a1, a2) as the input, I currently load an additional tokenizer by

```python
self.tokenizer_data_format = AutoTokenizer.from_pretrained("meta-llama/Meta-Llama-3-8B-Instruct", use_fast=True)
```